### PR TITLE
AA-816: update grade summary calculations

### PR DESCRIPTION
**Pull request by ciduarte**

### CODE REVIEW

File: src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx [hunk 1]
```diff
@@ -27,9 +27,18 @@ function GradeSummaryTable({
...
-    numPointsPossible > 0 ? ((numPointsEarned * assignmentWeight * 100) / numPointsPossible).toFixed(0) : 0
-  );
+  const calculateAssignmentTypeGrades = (points, assignmentWeight, numDroppable) => {
+    let dropCount = numDroppable;
+    while (dropCount) {
+      const lowestScore = Math.min(...points);
+      const lowestScoreIndex = points.indexOf(lowestScore);
+      points.splice(lowestScoreIndex, 1);
+      dropCount--;
+    }
```
Comment (Dillon-Dumesnil) [inline src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx]: I wonder if it's possible to just write a filter or something for this? This is totally fine though, just dreaming of the one line solution.

File: src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx [hunk 2]
```diff
@@ -27,9 +27,18 @@ function GradeSummaryTable({
 
   const footnotes = [];
 
-  const calculateWeightedGrade = (numPointsEarned, numPointsPossible, assignmentWeight) => (
-    numPointsPossible > 0 ? ((numPointsEarned * assignmentWeight * 100) / numPointsPossible).toFixed(0) : 0
-  );
+  const calculateAssignmentTypeGrades = (points, assignmentWeight, numDroppable) => {
+    let dropCount = numDroppable;
+    while (dropCount) {
```
Comment (Dillon-Dumesnil) [inline src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx]: Edge case likely, but might be good to consider to make sure dropCount isn't bigger than points.length. (in practice, should never happen, but someone could make a mistake or something /shrug)

**Review by Dillon-Dumesnil — APPROVED** [review]
Looks good to me.

File: src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx (hunk 2 above)
Comment (ciduarte) [reply src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx]: yeah Studio throws an error and doesn't allow the user to save if they've input a drop count larger than the total number of assignments, but anything is possible, so better to be defensive here 👍

File: src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx [hunk 3]
```diff
@@ -27,9 +27,22 @@ function GradeSummaryTable({
 
   const footnotes = [];
 
-  const calculateWeightedGrade = (numPointsEarned, numPointsPossible, assignmentWeight) => (
-    numPointsPossible > 0 ? ((numPointsEarned * assignmentWeight * 100) / numPointsPossible).toFixed(0) : 0
-  );
+  const calculateAssignmentTypeGrades = (points, assignmentWeight, numDroppable) => {
+    let dropCount = numDroppable;
+    while (dropCount && points.length && !(points.length < dropCount)) {
```
Comment (Dillon-Dumesnil) [inline src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx]: Switch the not to be a truthy statement.
`!(points.length < dropCount)` -> `points.length >= dropcount` or `dropcount < points.length`

Comment (Dillon-Dumesnil) [reply src/course-home/progress-tab/grades/grade-summary/GradeSummaryTable.jsx]: maybe a small comment above the `while` loop too explaining that this is dropping the lowest grades from the calculation.

File: src/course-home/data/__factories__/progressTabData.factory.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ Factory.define('progressTabData')
     },
     course_grade: {
       letter_grade: null,
-      percent: 0,
```
Comment (Dillon-Dumesnil) [inline src/course-home/data/__factories__/progressTabData.factory.js:16]: Why this switch?

**Review by Dillon-Dumesnil — APPROVED** [review]
Some small questions and nits, but yeah, in general looks good to me

File: src/course-home/data/__factories__/progressTabData.factory.js (hunk 4 above)
Comment (ciduarte) [reply src/course-home/data/__factories__/progressTabData.factory.js:16]: I wanted this to accurately reflect the rest of the data in the factory (this data should reflect a grade of 100% based on the assignment grades), but just realized I should input a letter grade as well. Gonna comb through this again and make sure everything lines up

**Pull request merged**